schema/connector-discovery-schema.json

```diff
@@ -0,0 +1,297 @@
+      "dependencies": {
+        "regex": {
+          "required": [
+            "capture_group"
+          ]
+        }
+      }
+    },
+    "actions": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/connector_level_action",
+        "description": "Each additional property is a connector level action."
+      }
+    },
+    "connector_level_action": {
+      "type": "object",
+      "properties": {
+        "url": {
+          "$ref": "#/definitions/link",
+          "description": "Link to the action."
+        },
+        "user_input": {
+          "type": "array",
+          "description": "Allows the user to enter data to be submitted in this action.",
+          "items": {
+            "$ref": "#/definitions/connector_action_user_input"
+          }
+        },
+        "fields": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/input_field",
+            "description": "Each additional property describes an input field."
+          }
+        },
+        "label": {
+          "type": "object",
+          "description": "Localised human-readable description of the action."
+        },
+        "completed_label": {
+          "type": "object",
+          "description": "A Localised message that says if the action was success in the backend."
+        },
+        "type": {
+          "$ref": "#/definitions/http_methods",
+          "description": "The HTTP method to use for submitting the action request."
+        },
+        "action_key": {
+          "type": "string",
+          "description": "A key by which client understand the way to submit the action. See Discovery in wiki"
+        },
+        "metadata": {
+          "type": "object",
+          "description": "Free form key-value pairs. Can contain additional metadata about the action, if anything is required by client."
+        }
+      },
+      "if": {
+        "properties": {
+          "action_key": {
+            "const": "USER_INPUT"
+          }
+        }
+      },
+      "then": {
+        "required": [
+          "user_input"
+        ]
+      },
+      "required": [
+        "url",
+        "type",
+        "label",
+        "action_key"
+      ]
+    },
+    "connector_action_user_input": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The field's ID, which will be used as the key when this field is submitted."
+        },
+        "label": {
+          "type": "object",
+          "description": "Localised field's label, which will be displayed to the user to describe the expected content of the field."
+        },
+        "format": {
+          "type": "string",
+          "description": "The format type of the user input."
+        },
+        "select": {
+          "type": "object",
+          "description": "If the format is 'select', choices for selecting should also be specified."
```

Comment:
If memory serves, the form value is the key and the label is the value.  It might be good to specify that.

@sumitzamadar can confirm in the boxer js code.

Reply:
Action level user inputs format supported by boxer:         

       "user_input": [
                  {
                    "id": "number",
                    "label": "Quantity",
                    "format": "number",
                    "range": [1, 5], /* range property is optional*/
                    "validation": ["required", "numeric"]
                  },
                  {
                    "id": "email",
                    "label": "E-mail",
                    "format": "email",
                    "validation": ["required", "email"]
                  },
                  {
                    "id": "telephone",
                    "label": "Telephone",
                    "format": "tel",
                    "validation": ["required", "phone"]
                  },
                  {
                    "id": "date",
                    "label": "Birthday",
                    "format": "date",
                    "validation": ["required", "date"]
                  },
                  {
                    "id": "range",
                    "label": "Points",
                    "format": "range",
                    "range": [0, 10] /* range property is optional; if not provided default will be 0-100 */
                  },
                  {
                    "id": "first_name",
                    "label": "First name",
                    "format": "text",
                    "validation": ["required"]
                  },
                  {
                    "id": "last_name",
                    "label": "Last name",
                    "format": "text",
                    "validation": ["required"]
                  },
                  {
                    "id": "comment_body",
                    "label": "Comment",
                    "format": "textarea",
                    "validation": ["required"]
                  },
                  {
                    "id": "favorite_color",
                    "label": "Favorite Color",
                    "format": "select",
                    "options": { "red": "Red", "green": "Green", "blue": "Blue", "orange": "Orange" },
                    "selected": "orange",
                    "validation": ["required"]
                  },
                  {
                    "id": "favorite_ice_cream",
                    "label": "Favorite Ice Cream",
                    "format": "radio",
                    "options": { "vanilla": "French Vanilla", "chocolate": "Triple Chocolate", "other": "Something else" },
                    "selected": "other",
                    "validation": ["required"]
                  },
                  {
                    "id": "radio_switch",
                    "label": "Switch",
                    "format": "radio",
                    "options": { "true": "On", "false": "Off" }
                  },
                  {
                    "id": "my_vehicles",
                    "label": "My Vehicles",
                    "format": "checkbox",
                    "options": { "Bike": "I have a bike", "Car": "I have a car" },
                    "selected": "Bike",
                    "validation": ["required"]
                  }
                ]

Reply:
Note: The Boxer code doesn't use connector-level-actions, however, we don't want to change "select" to work differently, so follow it for now.

Actually, since the select params are being defined by the discovery, we must specify locales like the labels and such.  Since we can't use the same schema as the card actions here, let's fix some of the problems (like order).

```
"select": [
        {
            "value": "some-uuid-or-something",
            "label": {
                "en-US": "...",
                "en-GB": "..."
            }
        }
    ]
}
```